debian/nfd.service

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2015,  Regents of the University of California,
+#                      Arizona Board of Regents,
+#                      Colorado State University,
+#                      University Pierre & Marie Curie, Sorbonne University,
+#                      Washington University in St. Louis,
+#                      Beijing Institute of Technology,
+#                      The University of Memphis.
+#
+# This file is part of NFD (Named Data Networking Forwarding Daemon).
+# See AUTHORS.md for complete list of NFD authors and contributors.
+#
+# NFD is free software: you can redistribute it and/or modify it under the terms
+# of the GNU General Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# NFD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# NFD, e.g., in COPYING.md file.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author: Eric Newberry <enewberry@email.arizona.edu>
+
+[Unit]
+Description=NDN Forwarding Daemon
+Documentation=man:nfd man:nfdc man:nfd-status
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+Environment=HOME=/var/lib/ndn/nfd
+ExecStart=/usr/bin/nfd --config /etc/ndn/nfd.conf
```

Comment:
I actually think /etc is OK under a couple conditions. Essentially, there should be no required setting in /etc and nfd should work fine out of the box without explicit configuration. This is very important for our ostree setup, but it's also good practice generally. Even in debian world it helps you not need admins to have to handle changes from debconf on updates.

There are 2 things I think are important to make this work:

1. nfd should run fine if there's no /etc/ndn/nfd.conf. If no settings in nfd.conf are required, then nfd should be fine if the file doesn't exist at all. This would be the ideal for our ostree world.

2. If you do want to ship a configuration file in /etc, it should not have any settings that are required for nfd to operate. I.e., any settings there should be commented out. That way, if a change wasn't merged in on upgrade (either ostree left it out because the original had been edited or debconf was run non-interactively), then nfd will still run correctly.

Is that possible with nfd? If that's the case, then I don't think you need the hack to move the configuration file on Endless.